src/lib/rpcClient.ts

```diff
@@ -95,12 +116,22 @@ export class RPCClient {
     prove: boolean
     height: number
   }) {
+    // Check if the data is a Uint8Array and convert it to a hex string if needed
     const query = Params.encodeAbciQuery({ method: Method.AbciQuery, params })
-    // Convert JsonRpcRequest to APIParams by passing it as a JSON string
-    const { response, uri } = await this.request('post', '', {
-      jsonRequest: JSON.stringify(query),
-    })
+
+    // Convert JsonRpcRequest to string
+    const queryString = JSON.stringify(query)
+
+    // Use the modified request method with jsonRequest parameter
+    const { response, uri } = await this.request<JsonRpcSuccessResponse>(
+      'post',
+      '',
+      undefined,
+      queryString
```

Comment:
I'm thinking about 
```
'post',
'',
{ ...query }
```
and we can completely remove `jsonRequest` from `request<T>`.

let me know if this is feasible or not

Reply:
actually let's go with your implementation since you already implemented the whole thing